set.mm

```diff
@@ -147154,6 +147148,405 @@ of an open range of nonnegative integers (of length equal to the length of
+
+    $( The zero length prefix is the empty set.  (Contributed by AV,
+       2-May-2020.) $)
+    pfx00 $p |- ( S prefix 0 ) = (/) $=
+      ( vs vl cc0 cop cpfx cdm wcel co c0 wceq cvv cn0 wa opelxp csubstr pfxval
+      cxp swrd00 cv syl6eq sylbi df-pfx dmmpt2 eleq2s wn cfv df-ov ndmfv syl5eq
+      ovex pm2.61i ) ADEZFGZHZADFIZJKZUQUMLMRZUNUMURHALHDMHNZUQADLMOUSUPADDEPIJ
+      ADLQADSUAUBBCLMBTZDCTEZPIFBCUCUTVAPUKUDUEUOUFUPUMFUGJADFUHUMFUIUJUL $.
+
+    $( A prefix of an empty set is always the empty set.  (Contributed by AV,
+       3-May-2020.) $)
+    pfx0 $p |- ( (/) prefix L ) = (/) $=
+      ( vs vl c0 cop cpfx cdm wcel co wceq cvv cn0 cxp wa opelxp csubstr pfxval
+      cc0 swrd0 cv syl6eq sylbi df-pfx ovex dmmpt2 eleq2s wn df-ov ndmfv syl5eq
+      cfv pm2.61i ) DAEZFGZHZDAFIZDJZUQUMKLMZUNUMURHDKHALHNZUQDAKLOUSUPDRAEPIDD
+      AKQRASUAUBBCKLBTZRCTEZPIFBCUCUTVAPUDUEUFUOUGUPUMFUKDDAFUHUMFUIUJUL $.
+  $}
+
+  ${
+    $d L x $.  $d S x $.  $d l s $.
+    $( Closure of the prefix extractor.  (Contributed by AV, 2-May-2020.) $)
+    pfxcl $p |- ( S e. Word A -> ( S prefix L ) e. Word A ) $=
+      ( vx vs vl cword wcel cpfx co c0 eleq1 wne cn0 cv wex n0 cc0 cop csubstr
+      wi cvv df-pfx elmpt2cl2 idi exlimiv sylbi wa pfxval swrdcl adantr eqeltrd
+      sylan2 wrd0 a1i pm2.61ne ) BAGZHZBCIJZUQHZKUQHZUSKUSKUQLUSKMZURCNHZUTVBDO
+      ZUSHZDPVCDUSQVEVCDVEVCUAEFUBNEORFOSTJBCIVDEFUCUDUEUFUGURVCUHUSBRCSTJZUQBC
+      UQUIURVFUQHVCABRCUJUKULUMVAURAUNUOUP $.
+
+    $d A x $.
+    $( Value of the prefix extractor as mapping.  (Contributed by AV,
+       2-May-2020.) $)
+    pfxmpt $p |- ( ( S e. Word A /\ L e. ( 0 ... ( # ` S ) ) )
+                    -> ( S prefix L ) = ( x e. ( 0 ..^ L ) |-> ( S ` x ) ) ) $=
+      ( cword wcel cc0 chash cfv cfz co wa cpfx cfzo cmpt cn0 wceq adantl nn0cn
+      syl cop csubstr cv caddc elfznn0 pfxval sylan2 simpl 0elfz simpr swrdval2
+      cmin syl3anc subid1d oveq2d elfzonn0 addid1d fveq2d mpteq12dva 3eqtrd ) C
+      BEZFZDGCHIZJKFZLZCDMKZCGDUAUBKZAGDGULKZNKZAUCZGUDKZCIZOZAGDNKZVJCIZOVDVBD
+      PFZVFVGQDVCUEZCDVAUFUGVEVBGGDJKFZVDVGVMQVBVDUHVEVPVRVDVPVBVQRDUITVBVDUJAB
+      CGDUKUMVEAVIVLVNVOVDVIVNQVBVDVHDGNVDVPVHDQVQVPDDSUNTUORVJVIFZVLVOQVEVSVKV
+      JCVSVJPFZVKVJQVJVHUPVTVJVJSUQTURRUSUT $.
+
+    $( Value of the subword extractor for left-anchored subwords.  (Contributed
+       by Stefan O'Rear, 24-Aug-2015.)  (Revised by AV, 2-May-2020.) $)
+    pfxres $p |- ( ( S e. Word A /\ L e. ( 0 ... ( # ` S ) ) )
+                    -> ( S prefix L ) = ( S |` ( 0 ..^ L ) ) ) $=
+      ( vx cword wcel cc0 chash cfv co wa cpfx cfzo cv cmpt cres pfxmpt wf wrdf
+      cfz adantr cuz wss elfzuz3 adantl fzoss2 syl feqresmpt eqtr4d ) BAEFZCGBH
+      IZTJFZKZBCLJDGCMJZDNBIOBUNPDABCQUMDGUKMJZAUNBUJUOABRULABSUAUMUKCUBIFZUNUO
+      UCULUPUJCGUKUDUECGUKUFUGUHUI $.
+
+    $d V x $.  $d W x $.
+    $( A prefix of a word is a function from a half-open range of nonnegative
```

Comment:
Is "half-open" the terminology used in set.mm ? (because for integer intervals, there is no notion of open or closed, but maybe it was decided in the past to mean "upper bound excluded" ?) So do as you see fit.

Reply:
Yes, this is the offical wording used in set.mm, see "5.5.7  Half-open integer ranges".